libdnf/hy-query.c

```diff
@@ -459,6 +519,35 @@ filter_release(HyQuery q, struct _Filter *f, Map *m)
 }
 
 static void
+filter_arch(HyQuery q, struct _Filter *f, Map *m)
+{
+    Pool *pool = dnf_sack_get_pool(q->sack);
+    Id match_arch_id = 0;
+    for (int mi = 0; mi < f->nmatches; ++mi) {
+        const char *match = f->matches[mi].str;
+        if (f->cmp_type & HY_EQ)
+            match_arch_id = pool_str2id(pool, f->matches[mi].str, 1);
```

Comment:
You can use `match` instead `f->matches[mi].str`
Why last argument is `1` ? Why do you want create new id? Memory is reallocated. And You lose information about `match` existence in pool. Don't do that.

Reply:
I think that this is correct. For arch filter the sensitivity for icase is counterproductive. Alse request that it should be not set is not good, because the python binding will have a problem.

Reply:
1. Why you do not use `match` constant instead of `f->matches[mi].str` ?